### PR TITLE
Feature/carddetails

### PR DESCRIPTION
**Pull request by varshini-id21**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -18,13 +18,14 @@ const routes: Routes = [
...
     { path: 'admin', component: AdminComponent },
     { path: 'home', component: LandingPageComponent },
+    { path: 'details/:category/:id', component: DetailsComponent }, 
     { path: '404', component: PagenotfoundComponent },
     { path: 'forgotpassword', component: ForgotpasswordComponent },
     { path: 'admindashboard', component: AdmindashboardComponent },
     { path: '**', redirectTo: '404', pathMatch: 'full' },
   ];
 @NgModule({
-  imports: [RouterModule.forRoot(routes,{useHash: true})],
```
Comment (faz2658) [inline src/app/app-routing.module.ts:27]: ?????

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,10 +1,10 @@
 <!-- <app-landing-page></app-landing-page> -->
 
 <app-navbar></app-navbar>
-<ng-container *ngIf="loading; else content">
+<!-- <ng-container *ngIf="loading; else content">
```
[the remark this answers is not in the document]
Comment (varshini-id21) [reply src/app/app.component.html:4]: getting error when i take that comment sir..

File: src/app/pages/culture/culture.component.ts [hunk 3]
```diff
@@ -1,103 +1,114 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit,OnDestroy } from '@angular/core';
+import { Router } from '@angular/router';
 import Card from 'src/app/models/card';
+import { CultureService,details } from 'src/app/services/culture-service.service';
```
Comment (faz2658) [inline src/app/pages/culture/culture.component.ts:4]: ??????

File: src/app/services/culture-service.service.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Environment, MockRoutes } from 'src/constants/routes';
+
+export interface details {
+  id : number;
+  title : string;
+  category : string;
```
Comment (faz2658) [inline src/app/services/culture-service.service.ts:8]: We are not writing interface in service , see where existing interfaces are written

File: src/app/services/details-service.service.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Environment, MockRoutes } from 'src/constants/routes';
```
Comment (faz2658) [inline src/app/services/details-service.service.ts:3]: This details-service is not needed

File: src/app/services/nature-service.service.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import { HttpClient } from '@angular/common/http';
+import { Environment, MockRoutes } from 'src/constants/routes';
+
+export interface details {
+  id : number;
+  title : string;
+  category : string;
+  subCategory : string;
+  thumbnail : string;
+  images : string[];
```
Comment (faz2658) [inline src/app/services/nature-service.service.ts:11]: same mistake

File: src/app/pages/details/details.component.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  currentSlideIndex: number = 0;
+
+  constructor(private route: ActivatedRoute, private detailsService: detailsService) {
+    this.route.params.subscribe(params => {
+      const id = Number(params['id']);
+      const category = params['category'];
+
+      switch (category) {
+        case 'nature':
+          this.detailsService.getNatureData().subscribe(data => {
```
Comment (faz2658) [inline src/app/pages/details/details.component.ts:33]: This is child class , not a dumb component . No service calls must be used

File: src/app/pages/culture/culture.component.ts (hunk 3 above)
Comment (faz2658) [inline src/app/pages/culture/culture.component.ts:4]: Wrong way,  service and interface imported from service

File: src/app/app.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (varshini-id21) [reply src/app/app.component.html]: solved this sir.

File: src/app/services/culture-service.service.ts (hunk 4 above)
Comment (varshini-id21) [reply src/app/services/culture-service.service.ts:8]: is it in card.ts sir?

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (faz2658) [inline src/app/app-routing.module.ts:27]: Why are we trying to remove existing functionality

**Review by faz2658 — CHANGES_REQUESTED** [review]

File: src/app/services/culture-service.service.ts (hunk 4 above)
Comment (varshini-id21) [reply src/app/services/culture-service.service.ts:8]: solved this have put this interface inside model/card.ts

File: src/app/services/nature-service.service.ts (hunk 6 above)
Comment (varshini-id21) [reply src/app/services/nature-service.service.ts:11]: solved this too

File: src/app/pages/culture/culture.component.ts (hunk 3 above)
Comment (varshini-id21) [reply src/app/pages/culture/culture.component.ts:4]: changes this one too...imported services from service and interface from interface

File: src/app/services/details-service.service.ts (hunk 5 above)
Comment (varshini-id21) [reply src/app/services/details-service.service.ts:3]: done this too now

**Pull request closed without merging**